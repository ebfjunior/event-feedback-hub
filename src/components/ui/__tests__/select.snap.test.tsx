import React from 'react';
import { render } from '@testing-library/react';
import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '../select';

// Reduce Lucide output noise
vi.mock('lucide-react', async () => {
	const actual = await vi.importActual<any>('lucide-react');
	return {
		...actual,
		ChevronDownIcon: (p: any) => <svg data-icon="chev-down" {...p} />,
		ChevronUpIcon: (p: any) => <svg data-icon="chev-up" {...p} />,
		CheckIcon: (p: any) => <svg data-icon="check" {...p} />,
	};
});

describe('ui/Select (snapshot)', () => {
	it('renders basic structure', () => {
		const { container } = render(
			<Select value="all">
				<SelectTrigger>
					<SelectValue />
				</SelectTrigger>
				<SelectContent>
					<SelectItem value="all">All</SelectItem>
					<SelectItem value="1">One</SelectItem>
				</SelectContent>
			</Select>,
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});


