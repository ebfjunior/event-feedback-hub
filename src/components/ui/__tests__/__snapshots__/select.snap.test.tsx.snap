// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ui/Select (snapshot) > renders basic structure 1`] = `
<button
  aria-autocomplete="none"
  aria-controls="radix-Â«r0Â»"
  aria-expanded="false"
  class="border-input data-[placeholder]:text-muted-foreground [&_svg:not([class*='text-'])]:text-muted-foreground focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive dark:bg-input/30 dark:hover:bg-input/50 flex w-fit items-center justify-between gap-2 rounded-md border bg-transparent px-3 py-2 text-sm whitespace-nowrap shadow-xs transition-[color,box-shadow] outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50 data-[size=default]:h-9 data-[size=sm]:h-8 *:data-[slot=select-value]:line-clamp-1 *:data-[slot=select-value]:flex *:data-[slot=select-value]:items-center *:data-[slot=select-value]:gap-2 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
  data-size="default"
  data-slot="select-trigger"
  data-state="closed"
  dir="ltr"
  role="combobox"
  type="button"
>
  <span
    data-slot="select-value"
    style="pointer-events: none;"
  >
    All
  </span>
  <svg
    aria-hidden="true"
    class="size-4 opacity-50"
    data-icon="chev-down"
  />
</button>
`;
